libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2125,6 +2180,106 @@ bool AP_Mission::contains_item(MAV_CMD command) const
     return false;
 }
 
+// reset the mission history to prevent recalling previous mission histories after a mission restart.
+void AP_Mission::reset_wp_history(void)
+{
+    for (uint8_t i = 0; i<AP_MISSION_MAX_WP_HISTORY; i++) {
+        _wp_index_history[i] = AP_MISSION_CMD_INDEX_NONE;
+    }
+    _resume_cmd.index = AP_MISSION_CMD_INDEX_NONE;
+    _flags.resuming_mission = false;
+    _repeat_dist = 0;
+}
+
+// store the latest reported position incase of mission exit and rewind resume
+void AP_Mission::update_exit_position(void)
+{
+    AP::ahrs().get_position(_exit_position);
```

Comment:
I wonder if we should check the return value of ahrs().get_position()

Reply:
Good spot.  I have added a check in.  The mission rewind function will now only rewind if there is a valid exit location.